connectd/connectd.h

```diff
@@ -27,6 +27,16 @@ struct gossip_state {
 	size_t off;
 };
 
+/*~ We need to know if we were expecting a pong, and why */
+enum pong_expect_type {
+	/* We weren't expecting a ping reply */
+	PONG_UNEXPECTED = 0,
+	/* We were expecting a ping reply due to ping command */
+	PONG_EXPECTED_COMMAND = 1,
+	/* We were expecting a ping reply due to ping timer */
```

Comment:
Nit: Don't we also frontrun commitments with a `ping`? I assume that'd also be bunched in with the timer.

Reply:
Not any more!  In the last release we just started pinging regularly, because Tor needed it to detect broken circuits.